web/app/jobs/app_uninstalled_job.rb

```diff
@@ -0,0 +1,23 @@
+class AppUninstalledJob < ActiveJob::Base
+  extend ShopifyAPI::Webhooks::Handler
+
+  class << self
+    def handle(topic:, shop:, body:)
+      perform_later(topic: topic, shop_domain: shop, webhook: body)
+    end
+  end
+
+  def perform(topic:, shop_domain:, webhook:)
+    shop = Shop.find_by(shopify_domain: shop_domain)
+
+    if shop.nil?
+      logger.error("#{self.class} failed: cannot find shop with domain '#{shop_domain}'")
+      return
+    end
+
+    shop.with_shopify_session do
```

Comment:
Do we need the shopify session to delete this? This should only affect the database rows.